diff --git a/src/browser/tab.ts b/src/browser/tab.ts
index 8d9ea5c..7a90bc5 100644
--- a/src/browser/tab.ts
+++ b/src/browser/tab.ts
@@ -39,6 +39,7 @@ export type TabSnapshot = {
   url: string;
   title: string;
   ariaSnapshot: string;
+  desc?: string;
   modalStates: ModalState[];
   consoleMessages: ConsoleMessage[];
   downloads: { download: playwright.Download, finished: boolean, outputFile: string }[];
@@ -196,10 +197,12 @@ export class Tab extends EventEmitter<TabEventsInterface> {
     let tabSnapshot: TabSnapshot | undefined;
     const modalStates = await this._raceAgainstModalStates(async () => {
       const snapshot = await (this.page as PageEx)._snapshotForAI();
+      const html = await this.page.content();
       tabSnapshot = {
         url: this.page.url(),
         title: await this.page.title(),
         ariaSnapshot: snapshot,
+        desc: `<truncated for brevity, length ${html.length}>`,
         modalStates: [],
         consoleMessages: [],
         downloads: this._downloads,
